bookwyrm/utils/cache.py

```diff
@@ -1,8 +1,15 @@
 """ Custom handler for caching """
+from typing import Any, Callable, Tuple, Union
+
 from django.core.cache import cache
 
 
-def get_or_set(cache_key, function, *args, timeout=None):
+def get_or_set(
+    cache_key: str,
+    function: Callable[..., Any],
+    *args: Tuple[Any, ...],
+    timeout: Union[float, None] = None
+) -> Any:
```

Comment:
Thanks, that's not something I could get from the code yet. Always nice to get rid of an `Any` annotation.

Reply:
Hmm, explicitly casting the result to `str` breaks a bunch of tests, so it looks like it is expected that it might also contain other types

Reply:
Ooh my bad, on a second look it appears this could be any of multiple things including various models objects. Not sure why I thought it was always a string!

Reply:
No problem, was a good test of the tests ;)